File: Konzol/C++/DataReader.cpp
#include "DataReader.h"
#include <fstream>
#include <sstream>
#include <iostream>

std::map<std::string, std::vector<int>> DataReader::BeolvasFajl(const std::string & fileName) {
    std::map<std::string, std::vector<int>> result;
    std::ifstream file(fileName);
    if (!file.is_open()) {
        std::cerr << "Hiba: Nem sikerült megnyitni a fájlt: " << fileName << std::endl;
        return result;
    }

    std::string line;
    bool isHeader = true;

    while (std::getline(file, line)) {
        if (isHeader) {
            isHeader = false;
            continue;
        }

        std::istringstream ss(line);
        std::string nyelv;
        std::getline(ss, nyelv, ';');

        std::vector<int> vizsgak;
        std::string value;
        while (std::getline(ss, value, ';')) {
            vizsgak.push_back(std::stoi(value));
        }

        result[nyelv] = vizsgak;
    }

    return result;
}

File: Konzol/C#/DataReader.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

public class DataReader
{
    public static Dictionary<string, List<int>> BeolvasFajl(string fileName)
    {
        var result = new Dictionary<string, List<int>>();

        using (var reader = new StreamReader(fileName, Encoding.UTF8))
        {
            string? line;
            bool isHeader = true;

            while ((line = reader.ReadLine()) != null)
            {
                if (isHeader)
                {
                    isHeader = false;
                    continue;
                }

                var parts = line.Split(';');
                var nyelv = parts[0];
                var vizsgak = parts.Skip(1).Select(int.Parse).ToList();
                result[nyelv] = vizsgak;
            }
        }

        return result;
    }
}